### PR TITLE
Add support for printing location

### PR DESCRIPTION
**Pull request by michaelkschmidt**
This currently isn't possible using standard format string, although it is included in the meta data

```elixir
config :logger, RingLogger
  location: true
```

Will prefix any message string with "module.function/arity : "

### CODE REVIEW

**Comment by adkron** [comment]
When utilizing the location option the log message would format as
`[debug] Module.function/airity : message`. After this it will format as
`[debug] Module.function/airity: message`. Also updates the Readme to
point to where the options are defined.

The reason for `location` as an option comes down to simplicity and
format. The module and function are allowed as part fore the metadata,
but the output is much harder to read, IMO.

`module=Module function=function/airity [debug] Hello`

I feel this is harder because it doesn't match the documentation format
of module, function, and airity. It also moves the level further into
the message. I think this hides the data and increases cognitive
overhead.

File: lib/ring_logger/client.ex [hunk 1]
```diff
@@ -97,12 +99,21 @@ defmodule RingLogger.Client do
   end
 
   def init(config) do
+    server_env = Application.get_env(:logger, RingLogger, [])
+    client_env = Application.get_env(:logger, __MODULE__, [])
```
Comment (fhunleth) [inline lib/ring_logger/client.ex]: The `server_env`/`client_env` lines aren't needed to support `:location`. Could you move those to a separate PR?

**Comment by adkron** [comment]
The point about `Logger.Formatter` is a good one. I don't know why I didn't think about that. I'll have to ponder on this for a bit.

**Comment by michaelkschmidt** [comment]
I agree this should go in logger/formatter.ex.  The general fix would be allow meta data to be specified in the format string.

The purpose of :location was to allow it to be used now since merging upstream will likely be time consuming.  It took literally 6 months last time for them to release our fixes for the proxy server in mix

**Comment by adkron** [comment]
With this patch in place, there is a bug. `Logger.bare_log(:debug, "my message")`

**Comment by fhunleth** [comment]
@michaelkschmidt Since we started exploring this change, I'm really thinking that it doesn't belong in an Elixir backend. I hadn't considered that before today, so I apologize for not pointing this out to you earlier. I also understand your concern that it takes some time to get changes into core Elixir. I still think that's the appropriate route. The possible other option is to make a custom formatter, but that pains me since prefixing messages with `$module.$function` seems trivial enough that `Logger.Formatter` should support it.

**Comment by adkron** [comment]
This might give us what we are after if you want to give it a 👍 https://github.com/elixir-lang/elixir/pull/7574

**Comment by fhunleth** [comment]
Since this isn't getting into core Elixir anytime soon, could one of you have a crack at #13 to support their recommended route?
